### PR TITLE
Add support for pm2 clustering

### PR DESCRIPTION
**Pull request by Tapppi**
Support [PM2 Clustering](http://pm2.keymetrics.io/docs/usage/cluster-mode/#graceful-reload) shutdown messages. If you think this should rather be a fork/higher level package/whatever feel free to reject and add your opinion.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Not sure it makes sense to have this here. This module is meant to catch any way a Node.js process exits, but this doesn't actually exit the process, it only indicates the process should exit. The dev has to manually call `process.exit` in the `message` event, which means it would be caught here anyways.

**Comment by Tapppi** [comment]
Actually, it does signify the process will exit after a set amount of time (4000ms default). It pretty much acts like a signal. The exit would be caught by the `exit` listener but that cannot support async handlers because the event loop will never be called again. The `beforeExit` on the other hand isn't called when you explicitly call `process.exit`. 
>['beforeExit' is not emitted for conditions causing explicit termination, such as process.exit()](https://nodejs.org/api/process.html#process_event_beforeexit)

File: index.js [hunk 1]
```diff
@@ -26,5 +26,12 @@ module.exports = function (cb) {
 		process.once('exit', exit);
 		process.once('SIGINT', exit.bind(null, true, 2));
 		process.once('SIGTERM', exit.bind(null, true, 15));
+
+		// PM2 Cluster shutdown message
+		process.on('message', function(msg) {
```
Comment (sindresorhus) [inline index.js]:
```js
process.on('message', function (msg) {
```

**Comment by sindresorhus** [comment]
Alright. I'll merge if you improve the code comment to succinctly reason about why it's there like you've done above.

**Pull request closed without merging**